### PR TITLE
#56 Improve Error Reporting

### LINKED ISSUE #56: Improve Error Reporting from CLI tool
In order to give the end user reasonable error messages, we need to add a wrapper for API-returned errors and check the HTTP status codes as well.

### PR DESCRIPTION
**Pull request by netme**
Closes #56 

Side-effect: removes couple of issues fro Codacy

### CODE REVIEW

**Comment by netme** [comment]
⚠️ to be rebased after #129 get merged

**Comment by fmueller** [comment]
👍

File: cli/src/main/java/de/zalando/zally/cli/CliException.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        this.title = title;
+        this.details = details;
+    }
+
+    @Override
+    public String getMessage() {
+        StringBuilder result = new StringBuilder();
+
+        if (title != null && !title.isEmpty()) {
+            result.append(getType() + title);
```
Comment (fmueller) [inline cli/src/main/java/de/zalando/zally/cli/CliException.java]: Minor: split this up into two append calls, no concatenation in append calls is best practice.

Comment (netme) [inline cli/src/main/java/de/zalando/zally/cli/CliException.java]: Thanks, fixed in 106682c

**Comment by netme** [comment]
And the good thing, we already have `Grade A` in our project :-)

**Comment by fmueller** [comment]
👍

**Pull request merged**